zerver/lib/user_groups.py

```diff
@@ -89,15 +89,20 @@ def has_user_group_access(
     if user_group.is_system_group:
         return False
 
+    can_edit_all_user_groups = user_profile.can_edit_all_user_groups()
```

Comment:
I am assuming the new realm-level setting will just be `Who can manage all user groups` and the permission will depend whether the user is member of the group irrespective of role, so have named this like this for now.